### DISCUSSION TITLE
[Feature Request] Add a Max Columns Compile Time Option

### DISCUSSION THREAD

**Discussion by jeffgazso** [andmarti1424/sc-im#817] [General]
Can a `-DMAXCOLUMNS` compile time option be added to complement the max rows option? Right now *sc-im* is capped at 702 columns per the README. Nearly all modern spreadsheets support at least 1024 columns and most support 16384 columns, either by default or as a configuration option. Adding a max columns option good for at least 16384 columns would bring *sc-im* to feature parity for large file support.

By the way, I know multiple sheet support was added in v0.8.3. Is the maximum number of sheets hard capped or is it simply limited by available memory?

**Comment by andmarti1424** [comment]
Hello. Do you really use that many columns?
You can redefine  #define ABSMAXCOLS   702 macro in sc.h if you need more.
The number of sheets is limited by memory, yes.

**Comment by jeffgazso** [comment]
@andmarti1424 

On very rare occasions yes, but not frequently. Usually if some `.csv` is that large I'm using *grep*, *awk* and friends to filter the data. I made the request out of concern for basic feature parity with other spreadsheet products in common use. Unfortunately, spreadsheets are treated as poor man's databases these days and they can grow to gigantic sizes.

I dug a bit deeper into the code and found `ABSMAXCOLS` after I submitted this feature request. Thank you for confirming I found the correct option. Would stress testing *sc-im* with 16384 column datasets and an unusually large number of sheets help the project or would I just be duplicating your efforts?

**Comment by TediusTimmy** [comment]
Note: yylex() and atocol() in lex.c have hard-coded assumptions that a column reference will only ever be two characters. Even if you change `ABSMAXCOLS`, other code has to change to reference those columns (and for them to display correctly).

Edit: and coltoa() in interp.c
